internal/imagedestination/impl/helpers.go

```diff
@@ -0,0 +1,19 @@
+package impl
+
+import (
+	"github.com/containers/image/v5/internal/private"
+	compression "github.com/containers/image/v5/pkg/compression/types"
+)
+
+// BlobMatchesRequiredCompression validates if compression is required by the caller while selecting a blob, if it is required
+// then function performs a match against the compression requested by the caller and compression of existing blob.
+func BlobMatchesRequiredCompression(options private.TryReusingBlobOptions, originalCompression *compression.Algorithm) bool {
+	if options.RequiredCompression == nil {
+		return true // no requirement imposed
+	}
+	return originalCompression != nil || (options.RequiredCompression.Name() == originalCompression.Name())
```

Comment:
```suggestion
	return originalCompression != nil && (options.RequiredCompression.Name() == originalCompression.Name())
```
AFAICS

Reply:
(I don’t generally think a unit test is required for a three-line function like this … but it would have helped here.)

Reply:
Oops, nice catch. Added a quick unit test for this.

Reply:
Modified tests a bit.

Reply:
```go
func TestBlobMatchesRequiredCompression(t *testing.T) {
	var opts private.TryReusingBlobOptions
	cases := []struct {
		requiredCompression  *compressionTypes.Algorithm
		candidateCompression *compressionTypes.Algorithm
		result               bool
	}{
		{&compression.Zstd, &compression.Zstd, true},
		{&compression.Gzip, &compression.Zstd, false},
		{&compression.Zstd, nil, false},
		{nil, &compression.Zstd, true},
	}

	for _, c := range cases {
		opts = private.TryReusingBlobOptions{RequiredCompression: c.requiredCompression}
		assert.Equal(t, BlobMatchesRequiredCompression(opts, c.candidateCompression), c.result)
	}
}
```